apps/backend/src/db/migrations/20240828160440_add_avatar_field.ts

```diff
@@ -0,0 +1,30 @@
+import { type Knex } from "knex";
+
+const TableName = {
+	FILES: "files",
+	USER_DETAILS: "user_details",
+} as const;
+
+const ColumnName = {
+	AVATAR: "avatar_file_id",
+	ID: "id",
+} as const;
+
+function up(knex: Knex): Promise<void> {
+	return knex.schema.alterTable(TableName.USER_DETAILS, (table) => {
+		table
+			.integer(ColumnName.AVATAR)
+			.unique()
+			.references(ColumnName.ID)
+			.inTable(TableName.FILES)
+			.onDelete("SET NULL");
```

Comment:
Should be CASCADE. Also move it to a separate variable

Reply:
I think it shouldn't be CASCADE because if a user deletes or updates their avatar, it will delete the record from the `user_details` table that is associated with that avatar because updating an avatar works like deleting a record from `files` and creating a new one, and then changing the `avatar_file_id` field in `user_details`. Is it better to change the avatar update logic?

Reply:
I think CASCADE works the other way around. When we delete the user, the avatar will be deleted, but the user will not be deleted when deleting avatar.